src/error.rs

```diff
@@ -49,7 +49,72 @@ pub struct Pos {
     path: String,
 }
 
+
+/// This type represents the location that an error occured
+#[derive(Debug)]
+pub struct ErrorMarker {
+    index: usize,
+    line: usize,
+    col: usize,
+} 
+
+impl ErrorMarker {
+    /// The byte index of the error
+    pub fn index(&self) -> usize {
+        self.index
+    }
+
+    /// The line of the error
+    pub fn line(&self) -> usize {
+        self.line
+    }
+
+    /// The column of the error
+    pub fn col(&self) -> usize {
+        self.col
+    }
+
+    fn from_marker(marker: &Marker) -> Self {
+        ErrorMarker {
+            col: marker.col(),
+            index: marker.index(),
+            line: marker.line(),
+        }
+    }
+}
+  
 impl Error {
+    /// Returns the ErrorMarker from the error if one exists.
+    /// Allows for the exact location of the error to be gathered.
```

Comment:
If you add an empty line above this one, the first line becomes a short and sweet summary in the API docs :)